### ISSUE TITLE
NK Wireless Controller

### ISSUE THREAD

**Issue by gervasito101** [ndeadly/MissionControl#793]
### Controller Name

NK Wireless Controller

### Current Controller Behaviour

Controller is not successful in pairing with the console

### Supported Features

- [ ] Rumble/Vibration
- [ ] Motion Controls
- [ ] Player Number Indicator
- [ ] Battery Level Reporting

**Comment by ndeadly** [comment]
Going to need some more information about this one, I have no idea what this controller is. A link to a vendor website, model number, picture...

Compatible bluetooth controllers should connect at the Change Grip/Order screen, even without button mapping. A controller that won't connect is likely:
- Using a problematic Bluetooth chipset
- Using Bluetooth LE (not currently supported, try the test build in my discord server under #testing-ble)
- A 2.4G wireless controller that requires a USB dongle (and is effectively a USB controller to the console)
- Having multiple connection modes, some of which may not be able to connect to the console

**Comment by gervasito101** [comment]
First of all, grateful for your prompt response. I'm sorry I can't give you much information as it was a gift, but I can provide you with the link where the product was purchased. Again thank you very much. Greetings. 
https://www.amazon.es/NK-Mando-para-Móvil-Inalámbrico/dp/B0BYK5885Y/ref=asc_df_B0BYK5885Y/?tag=googshopes-21&linkCode=df0&hvadid=646887993278&hvpos=&hvnetw=g&hvrand=5084245462393487832&hvpone=&hvptwo=&hvqmt=&hvdev=c&hvdvcmdl=&hvlocint=&hvlocphy=9047063&hvtargid=pla-2195774495459&mcid=452a9423fc34341f935f19c8e6185767&th=1

**Comment by pmnlla** [comment]
To me, it seems like a fairly generic Dinput controller, meaning it might just have a problematic Bluetooth chipset. Do you have a PS4/3 on hand? What's the pairing flow like? USB dongle or standard Bluetooth?

**Comment by gervasito101** [comment]
Tested on a PS4 it works without using a USB dongle; while on PS3 it only works using cable.

**Comment by ndeadly** [comment]
Oh, it's a PS4 clone. Yeah, if that doesn't connect then in all likelihood it's another one of the incompatible bluetooth chipsets.

**Comment by gervasito101** [comment]
Does all this mean that I will never be able to connect said controller to the ns through mission control? If there is no possible solution, I would like to ask you to close the thread so as not to bother you further and to thank you for everything.

**Comment by ndeadly** [comment]
Difficult to say. There seems to be a group of devices that are actually pairing with the console, but during opening of the hid connection are doing something that causes the console to immediately disconnect them. This case _may_ be resolvable through binary patching, but the only way that is going to happen is if somebody donates one of these controllers for me to use to debug the issue. There are also controllers that just straight up don't get detected by the bluetooth hardware on the Switch. For those there is likely no hope.

**Comment by gervasito101** [comment]
How can I get those controllers to you?  using a special program?  excuse my ignorance.

**Comment by ndeadly** [comment]
No, I meant the physical controller. I would need to debug on hardware while trying to pair an actual device.

**Comment by o0Zz** [comment]
Just in case it helps, it should work in USB mode with this sysmodule -> https://github.com/o0Zz/sys-con
